### DISCUSSION TITLE
xmake如何实现vs的项目依赖且依赖并不参与编译呢？

### DISCUSSION THREAD

**Discussion by augustheart** [xmake-io/xmake#2666] [Q&A] [answered]
目前xmake添加到add_deps的项目会强制参与编译，也就是最后会使用-l将依赖添加到链接目录。
但是实际项目中有些时候不需要依赖项目参与编译，但是需要它的另一个作用：在主项目编译之前编译依赖。

举例说：比如项目a.exe，它有一个组件b.dll，而b.dll在项目中是通过动态加载被a调用的，b本身没有导出函数（比如它就是一个纯资源dll） ，我编译a的时候，也希望它会同时编译b，但是由于b.dll无导出函数，在目前的情况下，最终会由于没有b.lib而出现编译错误。

**Comment by waruqi** [comment, marked as answer]
`add_deps("xxx", {inherit = false})` 文档有说明

**Reply by augustheart** [reply]
原来还有这个作用。add_deps的文档反复看过多遍，一直认为继承仅限include等配置项。
